src/jdk.compiler/share/classes/com/sun/tools/javac/comp/MemberEnter.java

```diff
@@ -285,7 +291,13 @@ public void visitVarDef(JCVariableDecl tree) {
         Type vartype = tree.isImplicitlyTyped()
                 ? env.info.scope.owner.kind == MTH ? Type.noType : syms.errType
                 : tree.vartype.type;
-        VarSymbol v = new VarSymbol(0, tree.name, vartype, enclScope.owner);
+        Name name;
+        if (Feature.UNNAMED_VARIABLES.allowedInSource(source) && (tree.mods.flags & UNNAMED) != 0) {
```

Comment:
Not 100% sure what this code does. E.g. shouldn't we just set the name of the variable as the name stored in the tree? It seems more economical if we could just set in the parser the name as empty (if preview features are enabled etc.) - and then all these special cases can disappear. After all, you still have the flag (and even then, do we need a flag? Can't we just see whether the name is empty to see that it's unnamed?)

Reply:
This is still needed because we want to have a different name for the Tree API (_) and the symbol (empty).